### ISSUE TITLE
scan_engine_connect.cc undeclared function ENONET breaks BSD build

### ISSUE THREAD

**Issue by jungle-boogie** [nmap/nmap#569]
Hello All,

nmap from master is failing to compile on my dragonfly BSD machine using gmake. 

```
LATFORM=\"x86_64-unknown-dragonfly4.7\" -DNMAPDATADIR=\"/usr/local/share/nmap\" -D_FORTIFY_SOURCE=2 -g -O2 -Wall -fno-strict-aliasing   portreasons.cc -o portreasons.o
g++ -c -I./liblinear -I./liblua -I./libdnet-stripped/include -I./libpcre  -I./nbase -I./nsock/include -DHAVE_CONFIG_H -DNMAP_NAME=\"Nmap\" -DNMAP_URL=\"https://nmap.org\" -DNMAP_PLATFORM=\"x86_64-unknown-dragonfly4.7\" -DNMAPDATADIR=\"/usr/local/share/nmap\" -D_FORTIFY_SOURCE=2 -g -O2 -Wall -fno-strict-aliasing   protocols.cc -o protocols.o
g++ -c -I./liblinear -I./liblua -I./libdnet-stripped/include -I./libpcre  -I./nbase -I./nsock/include -DHAVE_CONFIG_H -DNMAP_NAME=\"Nmap\" -DNMAP_URL=\"https://nmap.org\" -DNMAP_PLATFORM=\"x86_64-unknown-dragonfly4.7\" -DNMAPDATADIR=\"/usr/local/share/nmap\" -D_FORTIFY_SOURCE=2 -g -O2 -Wall -fno-strict-aliasing   scan_engine.cc -o scan_engine.o
g++ -c -I./liblinear -I./liblua -I./libdnet-stripped/include -I./libpcre  -I./nbase -I./nsock/include -DHAVE_CONFIG_H -DNMAP_NAME=\"Nmap\" -DNMAP_URL=\"https://nmap.org\" -DNMAP_PLATFORM=\"x86_64-unknown-dragonfly4.7\" -DNMAPDATADIR=\"/usr/local/share/nmap\" -D_FORTIFY_SOURCE=2 -g -O2 -Wall -fno-strict-aliasing   scan_engine_connect.cc -o scan_engine_connect.o
scan_engine_connect.cc: In function 'void handleConnectResult(UltraScanInfo*, HostScanStats*, std::__cxx11::list<UltraProbe*>::iterator, int, bool)':
scan_engine_connect.cc:308:10: error: 'ENONET' was not declared in this scope
     case ENONET:
          ^
gmake: *** [Makefile:109: scan_engine_connect.o] Error 1
```

I think this commit may have broken the build:
https://github.com/nmap/nmap/commit/fbc9ddae92265ca15d07a903412cfeee3915426e

Any suggestions?

**Comment by dmiller-nmap** [comment]
This is being addressed in #568 by surrounding the offending case with `#ifdef ENONET`. Sorry for the trouble, it should be merged in a day or so.

**Comment by jungle-boogie** [comment]
All better! Built without problems on 18d7a9195db6d6e4a7f2dd7aa9baf56f8018c294
